src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -201,144 +214,143 @@ protected TupleBatch fetchNextReady() throws IOException, DbException {
+                finalLineFound = true;
+                break;
               }
-              trailingEndInputStream.close();
-            } else {
-              finalLineFound = true;
-              finalBytePositionFound = maxByteRange;
+              dataChar = trailingEndInputStream.read();
+              finalBytePositionFound++;
             }
+            trailingEndInputStream.close();
+          } else {
+            finalLineFound = true;
+            finalBytePositionFound = maxByteRange;
+          }
 
-            /*
-             * If we found the new line, then reset the parser for this line. Otherwise, increase the byte overlap and
-             * the trailing range.
-             */
-            if (finalLineFound) {
-              long characterPositionAtBeginningOfRecord = record.getCharacterPosition();
-              InputStream completePartitionStream =
-                  source.getInputStream(adjustedStartByteRange, finalBytePositionFound);
-              BufferedReader reader =
-                  new BufferedReader(new InputStreamReader(completePartitionStream));
-              reader.skip(characterPositionAtBeginningOfRecord);
-              parser =
-                  new CSVParser(
-                      reader,
-                      CSVFormat.newFormat(delimiter).withQuote(quote).withEscape(escape),
-                      0,
-                      0);
-              iterator = parser.iterator();
+          /*
+           * If we found the new line, then reset the parser for this line. Otherwise, increase the byte overlap and the
+           * trailing range.
+           */
+          if (finalLineFound) {
+            long characterPositionAtBeginningOfRecord =
+                record == null ? 0 : record.getCharacterPosition();
+            InputStream completePartitionStream =
+                source.getInputStream(adjustedStartByteRange, finalBytePositionFound);
+            BufferedReader reader =
+                new BufferedReader(new InputStreamReader(completePartitionStream));
+            reader.skip(characterPositionAtBeginningOfRecord);
+            parser =
+                new CSVParser(
+                    reader,
+                    CSVFormat.newFormat(delimiter).withQuote(quote).withEscape(escape),
+                    0,
+                    0);
+            iterator = parser.iterator();
+            record = iterator.next();
+            if (nextRecordTruncated) {
               record = iterator.next();
-              if (truncatedQuotedField) {
-                record = iterator.next();
-              }
-              finishedReadingLastRow = true;
-            } else {
-              byteOverlap *= 2;
-              trailingStartByte += byteOverlap;
-              trailingEndByte += byteOverlap;
             }
+            nextRecordTruncated = true;
+            finishedReadingLastRow = true;
+          } else {
+            byteOverlap *= 2;
+            trailingStartByte += byteOverlap;
```

Comment:
Now I see why this was hard to catch. 

The point of this trailing stream is to find either a new line or the end of the file. If we skipped bytes which happened to have a new line, we'd still be ok because we'd eventually either reach another new line (from the next row) or the end of the file.  The ```completePartitionStream``` would just end up with unnecessary bytes to read.

Fixing it regardless of course. I was just stumped as to why my tests did not seem to catch this.